spec/migrations/migrate_shipping_taxes_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require_relative '../../db/migrate/20210224190247_migrate_shipping_taxes'
+
+describe MigrateShippingTaxes do
+  let(:migration) { MigrateShippingTaxes.new }
+
+  describe '#shipping_tax_category' do
+    it "creates a new shipping tax category" do
+      migration.shipping_tax_category
+
+      expect(Spree::TaxCategory.last.name).to eq I18n.t(:shipping)
+    end
+  end
+
+  describe '#create_shipping_tax_rates' do
+    let!(:zone1) { create(:zone_with_member) }
+    let!(:zone2) { create(:zone_with_member) }
+
+    before do
+      allow(migration).to receive(:instance_shipping_tax_rate) { 0.25 }
```

Comment:
Why not insert in the database and increase test coverage?

Reply:
There are some weird issues with `Spree::Config` in migrations (persisted in the `spree_preferences` table, but cached in the app layer), that why I initially went for stubbing this part in the specs. I notice various other  migrations that depend on `Spree::Config` values also go directly via `Spree::Preference` with named keys, eg: [this one](https://github.com/openfoodfoundation/openfoodnetwork/blob/76def3a74feb71995b8102da1e07522c57ae1ba3/db/migrate/20200404090436_remove_ga_cookies_preference.rb).

I'll take another look at it now. though and see what I can do...

Reply:
Ah... yeah it's made slightly more complicated by the fact that we're also _deleting_ that `Spree::Config` preference in this PR. So if we do `Spree::Config.set(shipping_tax_rate: 0.25)` in the spec (which would probably be the most reasonable way to set up the test in a way that mimics the real environment), we get an error saying the preference no longer exists. It'll still exist in the database, but it can't be set via `Spree::Config` now.

The alternative is to directly insert the preference key into the database (in the exact way the migration is already _expecting_), but I'm not sure there's much value in doing that? We'll just be reconfirming the pre-existing assumptions, right?

```diff
--      allow(migration).to receive(:instance_shipping_tax_rate) { 0.25 }
++      Spree::Preference.create(
++        key: '/spree/app_configuration/shipping_tax_rate',
++        value: 0.25,
++        value_type: "decimal"
++      )
```
Like this..?

Reply:
Ah, I see. We could do this and it would be marginally better but that may not be worth the effort. I'm okay with leaving it as is.